### PR TITLE
Edit profile info modal

### PR DESCRIPTION
**Pull request by ahsu1230**
This change contains the following:

* Rename address fields to include: `{name, street1, street2, city, province, country, postal}` as outlined in standards doc (can work with international addresses too)
* Backend for editing user info + tests
* Modal for editing user info and saves appropriately

### CODE REVIEW

**Comment by qwwqwwq** [comment]
Is this ready for review?

**Comment by ahsu1230** [comment]
mostly. I'm trying to see if I can consolidate parts of `editProfile.pug` to use `ng-repeat`. Other than that, it's about ready.

**Comment by ahsu1230** [comment]
now more ready

File: winterfell/test/testUserController.js [hunk 1]
```diff
@@ -58,12 +58,16 @@ describe('UserController', function() {
         .expect(http.OK, done);
   });
 
-  it('Delete user endpoint - success', function(done) {
+  it('Post user endpoint - success', function(done) {
     testSession
-        .del('/api/user')
+        .post('/api/user')
+        .send({firstname: 'Mister', lastname: 'Moose'})
         .expect(http.OK, done);
```
Comment (qwwqwwq) [inline winterfell/test/testUserController.js]: We should assert that it actually changed the user object in the database

**Comment by ahsu1230** [comment]
assert done (though already tested in the service test anyway)

**Comment by qwwqwwq** [comment]
LGTM pending merge conflict resolution (you will need to follow my pattern of doing the CSRF stuff, its pretty easy, just save the token once when you login, and then include it for all POST and DELETE requests)

**Pull request merged**